code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -1,116 +1,111 @@
 ////////////////////////////////////////////////////////////////////////////////
-/// Droppers.
+/// 								Droppers.								 ///
 ////////////////////////////////////////////////////////////////////////////////
+
 /obj/item/weapon/reagent_containers/dropper
 	name = "dropper"
 	desc = "A dropper. Transfers 5 units."
-	icon = 'icons/obj/chemical.dmi'
 	icon_state = "dropper"
+	item_state = "dropper"
 	amount_per_transfer_from_this = 5
-	possible_transfer_amounts = list(1,2,3,4,5)
+	possible_transfer_amounts = list(1, 2, 3, 4, 5)
 	volume = 5
 
 /obj/item/weapon/reagent_containers/dropper/on_reagent_change()
-	if(reagents.total_volume <= 0)
+	if(!reagents.total_volume)
 		icon_state = "[initial(icon_state)]"
 	else
 		icon_state = "[initial(icon_state)]1"
 
-/obj/item/weapon/reagent_containers/dropper/afterattack(obj/target, mob/user , flag)
-	if(!target.reagents)
-		return
-
-	if(reagents.total_volume)
+/obj/item/weapon/reagent_containers/dropper/attack(mob/living/M, mob/living/user, def_zone)
+	return
 
-		if(target.reagents.total_volume >= target.reagents.maximum_volume)
-			to_chat(user, "<span class='warning'>[target] is full.</span>")
+/obj/item/weapon/reagent_containers/dropper/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	var/to_transfer = 0
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(!reagents.total_volume)
 			return
-
-		if(!target.is_open_container() && !ismob(target) && !istype(target,/obj/item/weapon/reagent_containers/food) && !istype(target, /obj/item/clothing/mask/cigarette)) //You can inject humans and food but you cant remove the shit.
-			to_chat(user, "<span class='warning'>You cannot directly fill this object.</span>")
+		if(user != C)
+			visible_message("<span class='danger'>[user] begins attempting to squirt something into [C]'s eyes!</span>")

```

Comment:
This sounds really silly, like someone's straining really hard to empty an eyedropper. How about `"[user] begins to try and drip something into [C]'s eyes!"` instead? That makes more sense for a "slow" action.